File: src/main/java/bad/robot/radiate/ui/ExitOnEscape.java
package bad.robot.radiate.ui;

import java.awt.*;
import java.awt.event.AWTEventListener;
import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;

import static java.awt.event.KeyEvent.VK_ESCAPE;

class ExitOnEscape extends KeyAdapter implements AWTEventListener {
    @Override
    public void keyPressed(KeyEvent e) {
        if (e.getKeyCode() == VK_ESCAPE)
            System.exit(0);
    }

    @Override
    public void eventDispatched(AWTEvent event) {
        keyPressed((KeyEvent) event);
    }
}

File: src/main/scala/bad/robot/radiate/ui/ExitOnEscape.scala
package bad.robot.radiate.ui

import java.awt._
import java.awt.event.{AWTEventListener, KeyAdapter, KeyEvent}
import java.awt.event.KeyEvent.VK_ESCAPE

class ExitOnEscapeS extends KeyAdapter with AWTEventListener {
  override def keyPressed(e: KeyEvent) {
    if (e.getKeyCode == VK_ESCAPE) System.exit(0)
  }

  def eventDispatched(event: AWTEvent) {
    keyPressed(event.asInstanceOf[KeyEvent])
  }
}